<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>653</width>
    <height>534</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QwtPlot" name="plot" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tabPort">
       <attribute name="title">
        <string>Port</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Port:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="cbPortList">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="cbBaudRate">
              <property name="inputMethodHints">
               <set>Qt::ImhPreferNumbers</set>
              </property>
              <property name="editable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Baud Rate:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="pbReloadPorts">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>30</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>ðŸ”„</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QFrame" name="frame">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="QRadioButton" name="rbNoParity">
                 <property name="text">
                  <string>No Parity</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbOddParity">
                 <property name="text">
                  <string>Odd Parity</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbEvenParity">
                 <property name="text">
                  <string>Even Parity</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="verticalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>2</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame_3">
              <layout class="QVBoxLayout" name="verticalLayout_7">
               <item>
                <widget class="QRadioButton" name="rb8Bits">
                 <property name="text">
                  <string>8 bits</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb7Bits">
                 <property name="text">
                  <string>7 bits</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb6Bits">
                 <property name="text">
                  <string>6 bits</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb5Bits">
                 <property name="text">
                  <string>5 bits</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame_4">
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <widget class="QRadioButton" name="rb1StopBit">
                 <property name="text">
                  <string>1 Stop Bit</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb2StopBit">
                 <property name="text">
                  <string>2 Stop Bit</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="verticalSpacer_4">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>2</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame_2">
              <layout class="QVBoxLayout" name="verticalLayout_6">
               <item>
                <widget class="QRadioButton" name="rbNoFlowControl">
                 <property name="text">
                  <string>No Flow Control</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbHardwareControl">
                 <property name="text">
                  <string>Hardware Control</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbSoftwareControl">
                 <property name="text">
                  <string>Software Control</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="verticalSpacer_5">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>2</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>10000</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="CustomCheckableButton" name="pbOpenPort">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="text">
             <string>Open</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pbSkipByte">
            <property name="text">
             <string>Skip Byte</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabDataFormat">
       <attribute name="title">
        <string>Data Format</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Number Format:</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QRadioButton" name="rbUint8">
             <property name="text">
              <string>uint8</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QRadioButton" name="rbUint32">
             <property name="text">
              <string>uint32</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QRadioButton" name="rbUint16">
             <property name="text">
              <string>uint16</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QRadioButton" name="rbInt16">
             <property name="text">
              <string>int16</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QRadioButton" name="rbInt32">
             <property name="text">
              <string>int32</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QRadioButton" name="rbInt8">
             <property name="text">
              <string>int8</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <spacer name="verticalSpacer_6">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>445</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabPlot">
       <attribute name="title">
        <string>Plot</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>370</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Number Of Samples:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spNumOfSamples">
              <property name="keyboardTracking">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <number>2</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="value">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>653</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionFileLoadProfile"/>
    <addaction name="actionFileSaveProfile"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelpDataFormats"/>
    <addaction name="actionHelpAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPause"/>
   <addaction name="actionClear"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pause Plotting</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionHelpDataFormats">
   <property name="text">
    <string>Data Formats</string>
   </property>
  </action>
  <action name="actionHelpAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionFileLoadProfile">
   <property name="text">
    <string>Load Profile</string>
   </property>
  </action>
  <action name="actionFileSaveProfile">
   <property name="text">
    <string>Save Profile</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QWidget</extends>
   <header>qwt_plot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CustomCheckableButton</class>
   <extends>QPushButton</extends>
   <header>customcheckablebutton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
